### DISCUSSION TITLE
Alarm pannel not available

### DISCUSSION THREAD

**Discussion by jvilafe** [guerrerotook/securitas-direct-new-api#101] [Q&A]
Hi guys
 I can install this component, add integration and setup integration component...I can setup my user/password/code/country and securitas direct sends auth code to my cell phone, so, I think that all is ok, but...
 I can select alarm card because there is no alarm entity from this integration...

regards

**Comment by mikehughesc** [comment]
Hi - did you get a solution for this?  I have the same problem.

**Comment by Xalfen** [comment]
Hello,
Same issue here, no entity or sensor detected, can't add anything.

**Comment by jvilafe** [comment]
Hi

yes, my issue was with login credentials to securitas http service. My account was blocked on the securitas/verisure web... I tried to access the url services and I can't!! So, I made the unlock process on the securitas web and retry from HA integration and it works! I can add the alarm pannel.

**Comment by mikehughesc** [comment]
How do the login credentials work?  The HA config fle wants a username, password, and "code" - what is that code?  I have a username and password for the securitas website, but the only code I can think of is the code name that you give when you've set off the alarm accidentally! - is that it?  I tried that and still don't get any alarm panel (unless I'm looking in the wrong place for the panel).

**Comment by Xalfen** [comment]
I think it’s the pin code to turn on off the alarm ?

Le mer. 15 févr. 2023 à 23:03, mikehughesc ***@***.***> a
écrit :

**Comment by jvilafe** [comment]
Hi, 

the code is only a PIN to unlock de alarm pannel in the HA integration. This code is NOT provided by Securitas Direct!! You can use any PIN (1234) in this field; Then, you can use this code in the Home Assistant Alarm pannel to unlock this pannel .

**Reply by mikehughesc** [reply]
I tried a 4 digit PIN, but it failed the config file verification before restarting HA. Can't remember the exact message (I'll try again a bit later and copy it here if needed) but it implied that a numeric value for code was not allowed. I tried also putting a numeric value in quotes "1234" and that passed the validation test, but still no new entities or devices.

**Reply by jvilafe** [reply]
no, this is the pin to lock/unlock the access to alarm pannel on HA.

**Comment by Xalfen** [comment]
Checked the logs, found:
```
2023-02-16 09:18:46.929 ERROR (MainThread) [homeassistant.config_entries] Error unloading entry MYUSERNAME for sensor
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 533, in async_unload
    result = await component.async_unload_entry(hass, self)
  File "/usr/src/homeassistant/homeassistant/components/sensor/__init__.py", line 131, in async_unload_entry
    return await component.async_unload_entry(entry)
  File "/usr/src/homeassistant/homeassistant/helpers/entity_component.py", line 181, in async_unload_entry
    raise ValueError("Config entry was never loaded!")
ValueError: Config entry was never loaded!
```

So that would explain why the sensor and entities aren't showing up

**Comment by mikehughesc** [comment]
I tried:

securitas:
  username: MYUSERNAME
  password: MYPASSWORD
  code: 1234
  country: FR

Clicking "check configuration" before restart gives:

Configuration invalid!
Invalid config for [securitas]: expected str for dictionary value @ data['securitas']['code']. Got 1234. (See /config/configuration.yaml, line 34). 
           
Putting in code: ABCD passes the validation test - does this code have to be alphanumeric?

But in any case, when I try tro add an alarm panel to the dashboard, I get an error:

The provided value for 'entity' is not supported by the visual editor. We support (entity ID (domain.entity)) but received ("").

Also I get the same log entries as Xalfen above.

**Comment by SirLouen** [comment]
This is a general problem. Probably the API has been updated and the entity is return than error so it cannot be used for the panel

Maybe @guerrerotook is looking to this issue.

**Comment by SirLouen** [comment]
Found the solution here: https://github.com/guerrerotook/securitas-direct-new-api/issues/107

You have to edit just these two files:

custom_components/securitas/__init__.py
custom_components/securitas/config_flow.py

and change the `async_forward_entry_setups` to `async_setup_platforms`

**Reply by guerrerotook** [reply]
Will take a look at this tomorrow!

**Reply by SirLouen** [reply]
I'm not into this coding, but this may help you out finding the culprit. From what I've read, it is not as easy as just changing the function. You need to perform some extra steps to make it work.
https://developers.home-assistant.io/docs/config_entries_index/

```
# Use `hass.async_create_task` to avoid a circular dependency between the platform and the component
```

**Reply by guerrerotook** [reply]
That was because there is a warning for the next release, but I can revert and investigate more that function.

**Reply by SirLouen** [reply]
> That was because there is a warning for the next release, but I can revert and investigate more that function.

Found the issue, was simpler than expected I've sent a pull request to solve this

**Comment by guerrerotook** [comment]
I created this new release reverting the changes to make sure that works for now until I fix the issue. 

https://github.com/guerrerotook/securitas-direct-new-api/releases/tag/V2.1.0.1 

Please update as soon as possible.

**Comment by mikehughesc** [comment]
OK that's great - I am now able to add the alarm panel! - many thanks!

Still two problems, however.

The first is that as described above, it is impossible to set the code to a numeric PIN - if you do that, the configuration file validation fails with an error message.  So I temporarily set it to A123 and that works - but it means you have to have a keyboard to type the code in, you can't use the numeric keypad built into the alarm panel!

The second problem is that I was able to set the "alarm away", but disarm didn't work - it gave the following error.  It looks like that might be related to the code not being numeric?

![Screenshot 2023-02-18 181756](https://user-images.githubusercontent.com/66436378/219879293-ef7ec032-5ad9-422a-82b5-8cbeb33ae9e2.png)

**Reply by guerrerotook** [reply]
I need to keep working on the service, because is not finished, let me take a look!

**Comment by mikehughesc** [comment]
FYI I just found this in my HA logs - not sure if you're already aware?

 Logger: homeassistant.helpers.frame
Source: helpers/frame.py:77
First occurred: 28 February 2023 at 15:27:39 (1 occurrences)
Last logged: 28 February 2023 at 15:27:39
Detected integration that called async_setup_platforms instead of awaiting async_forward_entry_setups; this will fail in version 2023.3. Please report issue to the custom integration author for securitas using this method at custom_components/securitas/__init__.py, line 186: hass.config_entries.async_setup_platforms(entry, PLATFORMS)

**Reply by SirLouen** [reply]
Yes, this is the problem in the previous version, before the rollback. I think that @guerrerotook has not found yet the solution to introduce async_forward_entry_setups in the component. Although I have not updated yet. Maybe the latest two updates solve this, not 100% sure. What version are you using currently?

EDIT: I'm checking https://github.com/guerrerotook/securitas-direct-new-api/commit/5c21b30b8abcdd6882619ba9c820dcb1af43450b

And that was modified, so you must be using an old version. I'm going to update now

**Comment by gerard-marti** [comment]
Hi, installing securitas component and setting the information I have some issues:
- First one, If I put code as a numeric value i get the same error as @mikehughesc 
- Second error, if i change that code to "1234" for example, and I restart haos, the panel is not visible due this error:
![imagen](https://user-images.githubusercontent.com/14344504/222705365-6d766bd8-0be0-4981-9c87-ad6547f37d9a.png)

Some help here? Thanks!

**Reply by guerrerotook** [reply]
Normally the alarm-panel automatically appear in the UI if you add the integration. Do you have it?

**Reply by gerard-marti** [reply]
Nope, it does not appear

**Reply by guerrerotook** [reply]
Do you have anything in the logs, did you do the process of two factor authentication?

**Reply by gerard-marti** [reply]
Sorry, I did nothing about 2-fa, what I have to do?

**Reply by SirLouen** [reply]
> Sorry, I did nothing about 2-fa, what I have to do?

When you install the HACS extension, then you add it as an integration and you will go under a series of login forms, including a 2FA with Securitas Direct sending to you an SMS code to confirm everything.

**Reply by gerard-marti** [reply]
mmm I didn't go through any login form, maybe I did something wrong? I just put the login conf inside the yaml

**Reply by gerard-marti** [reply]
Ok, Solved it, thanks for the info, but arming the alarm gives me this error: 
![imagen](https://user-images.githubusercontent.com/14344504/222747341-47b76576-2e64-45c2-b92e-e6d8cf5910ce.png)

**Reply by SirLouen** [reply]
In fact, I'm not using the config file... Never sure why it could be useful. I have it registered through the integration form.

Maybe the configuration creates the integration automatically @guerrerotook ?

**Reply by guerrerotook** [reply]
Because the integration used config flow, regardless of how you add the integration it always will go to the config registry. So it's doesn't matter how you add it.

**Reply by SirLouen** [reply]
> Because the integration used config flow, regardless of how you add the integration it always will go to the config registry. So it's doesn't matter how you add it.

How do you pass the 2FA with the config file? When you add the integration with the config file previously set up, it bypasses the config form and it goes straightly to the 2FA?

**Reply by guerrerotook** [reply]
> > Because the integration used config flow, regardless of how you add the integration it always will go to the config registry. So it's doesn't matter how you add it.
> 
> How do you pass the 2FA with the config file? When you add the integration with the config file previously set up, it bypasses the config form and it goes straightly to the 2FA?

You don't, in the UI it should be an entry saying that you need to finish the configuration for the integration.

**Reply by SirLouen** [reply]
I tend not to config much through files so I tend to forget when I did it or not after a few months, I feel that the double option is more messy than just 1 alternative.

**Reply by gerard-marti** [reply]
> Ok, Solved it, thanks for the info, but arming the alarm gives me this error: ![imagen](https://user-images.githubusercontent.com/14344504/222747341-47b76576-2e64-45c2-b92e-e6d8cf5910ce.png)

So how can I solve this? Thanks!

**Reply by gerard-marti** [reply]
Solved, I was calling arm on house, and I only have alarm on leave. Thanks!

**Comment by mikehughesc** [comment]
I'm still getting the same errors as reported earlier - can anyone help please?

1. It is impossible to set the code to a numeric PIN - if you do that, the configuration file validation fails with an error message. So I temporarily set it to A123 and that works - but it means you have to have a keyboard to type the code in, you can't use the numeric keypad built into the alarm panel!

2. I was able to set the "alarm away", but disarm didn't work - it gave "invalid literal for int()" as pre screenshto in my previous post. It looks like that might be related to the code not being numeric? 

(And yes, I went through the proper 2FA initialisation process)

**Reply by guerrerotook** [reply]
did you try to add the configuration without code?

**Reply by mikehughesc** [reply]
No - I thought the code entry in the config was required in order to define the PIN for setting/unsetting the alarm.  If I remove the code from the config file, how will it know what code I am going to use to set/unset the alarm?

**Reply by SirLouen** [reply]
You don't set the alarm with a code

To unset the alarm from Lovelace, you simply put the code in the keyboard panel
To unset the alarm automatically with an automation, there is a input to put such code also when you create the automation.

**Reply by gerard-marti** [reply]
If you quit code field of yaml configuration, it marks it as invalid saying code is mandatory

**Reply by mikehughesc** [reply]
"To unset the alarm from Lovelace, you simply put the code in the keyboard panel" - OK but what code?  I understood that the code to be entered in the panel was the code you previously specified in the config file!  What other code is there? - my real life physical alarm is unset with a proximity sensor, not a code.

As @jvilafe said in his post a few weeks ago:  "the code is only a PIN to unlock de alarm pannel in the HA integration. This code is NOT provided by Securitas Direct!! You can use any PIN (1234) in this field; Then, you can use this code in the Home Assistant Alarm pannel to unlock this pannel."

**Reply by SirLouen** [reply]
> "To unset the alarm from Lovelace, you simply put the code in the keyboard panel" - OK but what code? I understood that the code to be entered in the panel was the code you previously specified in the config file! What other code is there? - my real life physical alarm is unset with a proximity sensor, not a code.

Ok, you have the old version of Securitas Alarm the `Verifast`

In that case I cannot tell. I tried to configure my old alarm but I did not finish. My new alarm the `Moonshot` version does have the panel with the code. This is the code you need.

Theorically there is a code you can use to disarm your alarm in the head (if you see there are like some buttons with numbers around the head of the alarm). If you contact Securitas, they will give you such code. Generally that code is set up the first day the technician goes to your home.

**Reply by mikehughesc** [reply]
If you remove the code from the config, then HA refuses to restart -

Configuration invalid!
Invalid config for [securitas]: expected str for dictionary value @ data['securitas']['code']. Got None. (See /config/configuration.yaml, line 38). 

And if you put in a numeric code then you get an error, and if you put in an alphanumeric code then it doesn't work (and you can't enter alphabetic characters on the keypad in the panel anyway).

**Reply by guerrerotook** [reply]
Let me make the code optional and push for an update! Will keep you posted.

**Reply by gerard-marti** [reply]
Thanks!

**Reply by mikehughesc** [reply]
Can you also make it allow numeric-only ?

**Reply by SirLouen** [reply]
> Let me make the code optional and push for an update! Will keep you posted.

So the code doesn't have anything to do with the real alarm code of the alarm?

**Reply by guerrerotook** [reply]
> > Let me make the code optional and push for an update! Will keep you posted.
> 
> So the code doesn't have anything to do with the real alarm code of the alarm?

No, it's a Home Assistant thing.

**Reply by guerrerotook** [reply]
> Can you also make it allow numeric-only ?

It's already a string so that can be a pin of four numbers, there is no need to force numeric only. There are other that had other codes.

**Reply by guerrerotook** [reply]
This new release removes the requirement for the code. Please let me know if it works. 

https://github.com/guerrerotook/securitas-direct-new-api/releases/tag/v2.1.0.4

**Reply by mikehughesc** [reply]
No, you misunderstood - if you enter four numerics you get an error.  It only works if you enter at least one alphabetic character.  If I have:

  code: 1234

then when I check the config file I get:

Configuration invalid!

Invalid config for [securitas]: expected str for dictionary value @ data['securitas']['code']. Got 1234. (See /config/configuration.yaml, line 38).

**Reply by guerrerotook** [reply]
I have a pin like this, did you already configure the integration? You can manually modify the pin. I explain how in the release description.

**Reply by mikehughesc** [reply]
Yes, I have it all configured (with code A123) and the panel works to arm the alarm but not to disarm it.  What am I doing wrong? - I have:

securitas:
  username: 9999999
  password: mypasswordxxxxx
  code: 123A
  country: FR
  check_alarm_panel: false

and that works - but if I put code=1234 then it refuses to restart HA.  (all lines after securitas are indented 2 spaces)

PS - sorry what's the release description? - I cannot find any documentation for this except these discussions.

**Reply by guerrerotook** [reply]
> Yes, I have it all configured (with code A123) and the panel works to arm the alarm but not to disarm it. What am I doing wrong? - I have:
> 
> securitas: username: 9999999 password: mypasswordxxxxx code: 123A country: FR check_alarm_panel: false
> 
> and that works - but if I put code=1234 then it refuses to restart HA. (all lines after securitas are indented 2 spaces)

If you know how to modify the code, can you paste this on line 67 in the __init__.py file for the integration?

vol.Optional(CONF_COUNTRY, default="ES"): vol.Any(str, int),

This will allow numbers and string and please try again. Thanks!

**Reply by mikehughesc** [reply]
OK, happy to try that - do you mean the file called --init--.py (with underscores instead of hyphens)?

Also there are two copies of that file in the zip file for the custom component - I just downloaded it again to make sure - 

\custom_components\securitas\\--init--.py
\custom_components\securitas\securitas_direct_new_api\\--init--.py

Do I edit both of them?

**Reply by SirLouen** [reply]
> No, it's a Home Assistant thing.

Where is saved the pin inputted through the form, not the config file?
